e2etest/GuestProxyAgentTest/LinuxScripts/InstallGuestProxyAgentExtension.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+zipFile=$devExtensionSas   
+
+echo "Starting install guest proxy agent extension script" 
+timeout=300
+elpased=0
+echo "Get Extension Folder and Version"
+while :; do
+    directories=$(find /var/lib/waagent -type d -name '*Microsoft.CPlat.ProxyAgent.ProxyAgentLinux*')
+    if [ $(echo "$directories" | wc -l) -eq 1 ]; then
+        for dir in $directories; do 
+            PIRExtensionFolderPath=$dir
+            echo "PIR extension folder path and contents" $PIRExtensionFolderPath
+            PIRExtensionFolderZip="${PIRExtensionFolderPath//-/__}.zip"
+            echo "PIRExtensionFolderZip:$PIRExtensionFolderZip"
+        done 
+        break
+    fi
+    ((elapsed += interval))
+    if [[ $elapsed -ge $timeout ]]; then
+        echo "Timeout reached. Exiting the loop."
+        break
+    fi
+    sleep 5
+done 
+PIRExtensionVersion=$(echo "$PIRExtensionFolderPath" | grep -oP '(\d+\.\d+\.\d+)$')
+echo "PIRExtensionVersion=$PIRExtensionVersion"
+proxyAgentVersion="$(eval "$PIRExtensionFolderPath/ProxyAgent/ProxyAgent/azure-proxy-agent --version")"
+echo "proxy agent version: $proxyAgentVersion"
+statusFolder=$(find "$PIRExtensionFolderPath" -type d -name 'status')
+echo "Status Directory: $statusFolder"
+
+echo "Delete status file of PIR version" 
+rm -rf $statusFolder/*
+
+echo "detecting os and installing jq" 
+os=$(hostnamectl | grep "Operating System")
+echo "os=$os"
+if [[ $os == *"Ubuntu"* ]]; then
+    for  i in {1..3}; do
+        echo "start installing jq via apt-get $i"
+        sudo apt update
+        sudo apt-get install -y jq
+        sleep 10
+        install=$(apt list --installed jq)
+        echo "install=$install"
+        if [[ $install == *"jq"* ]]; then
+            echo "jq installed successfully"
+            break
+        fi
+    done
+else
+    for  i in {1..3}; do
+        echo "start installing jq via yum $i"
+        sudo yum -y install jq
+        sleep 10
+        install=$(yum list --installed jq)
+        echo "install=$install"
+        if [[ $install == *"jq"* ]]; then
+            echo "jq installed successfully"
+            break
+        fi
+    done
+fi
+
+echo "Check that status file is success with 5 minute timeout"
```

Comment:
Not sure if this status check is necessary since PIR status is already checked in GuestProxyAgentExtensionValidation. Could remove lines 34-99

Reply:
same for windows..